### PR TITLE
kevm_pyk: pretty print KInner tokens as hex.

### PR DESCRIPTION
**Pull request by anvacaru**
closes: https://github.com/runtimeverification/evm-semantics/issues/1792
- Implements a `pretty_print` function in the `KEVM` class in kevm-pyk that prints elements of sorts `Int` and `Bytes` in hexadecimal format in a KCFG.
- adds a new `unit/test_kevm.py` file to test the functionality.
- added a `use_hex` argument to the `KEVM.__init__` method that enables the hex encoding in pretty printing.

image

### CODE REVIEW

File: kevm-pyk/src/kevm_pyk/kevm.py [hunk 1]
```diff
@@ -316,6 +318,30 @@ def _add_account_invariant(account: KApply) -> list[KApply]:
...
+        def _to_hex(kast: KInner) -> KInner:
+            if type(kast) is KToken and kast.sort == INT:
+                return KToken(hex(int(kast.token)), INT)
+            if type(kast) is KToken and kast.sort == BYTES:
+                return KToken('0x' + pretty_bytes(kast).hex(), BYTES)
+            return kast.map_inner(_to_hex)
+
+        if isinstance(kast, KToken):
+            return _to_hex(kast)
+        return kast.map_inner(_to_hex)
```
Comment (tothtamas28) [inline kevm-pyk/src/kevm_pyk/kevm.py:343]:
```suggestion
        return kast.top_down(_to_hex)
```

File: kevm-pyk/src/kevm_pyk/kevm.py [hunk 2]
```diff
@@ -316,6 +318,30 @@ def _add_account_invariant(account: KApply) -> list[KApply]:
...
+            kast = KEVM.kinner_to_hex(kast)
+        return super().pretty_print(kast, unalias=unalias, sort_collections=sort_collections)
+
+    @staticmethod
+    def kinner_to_hex(kast: KInner) -> KInner:
+        """
+        Converts values within a KInner object of sorts `INT` or `BYTES` to hexadecimal representation.
+        """
+
+        def _to_hex(kast: KInner) -> KInner:
```
Comment (tothtamas28) [inline kevm-pyk/src/kevm_pyk/kevm.py:334]:
```suggestion
        def to_hex(kast: KInner) -> KInner:
```

**Review by tothtamas28 — APPROVED** [review]

File: kevm-pyk/src/kevm_pyk/kevm.py (hunk 2 above)
Comment (anvacaru) [reply kevm-pyk/src/kevm_pyk/kevm.py]: Done!

File: kevm-pyk/src/kevm_pyk/kevm.py (hunk 1 above)
Comment (anvacaru) [reply kevm-pyk/src/kevm_pyk/kevm.py]: Done!

**Comment by anvacaru** [comment]
@ehildenb Added a new `use_hex` flag in the constructor.

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**